### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by uellue**
Initial version
    
  * pyproject.toml and such
  * Tests
  * Functionality
  * Examples
  * Continue  README

### CODE REVIEW

File: src/sparseconverter/__init__.py [hunk 1]
```diff
@@ -0,0 +1,953 @@
...
+    return cheapest_key
+
+
+def for_backend(arr: ArrayT, backend: Optional[ArrayBackend], strict: bool = True) -> ArrayT:
+    converter = get_converter(get_backend(arr), backend, strict)
+    # print(source_backend, backend, converter)
+    return converter(arr)
+
+
+def check_shape(arr: ArrayT, shape: Iterable[int]) -> bool:
```
Comment (sk1p) [inline src/sparseconverter/__init__.py:857]: Needs a docstring - just looking at the signature, I would assume that is returns `False` in the incompatible case, but it raises a `ValueError` instead. I think it should explicitly have a return type of `None`, or consistently return `True`/`False`.

**Review by sk1p — COMMENTED** [review]
General idea and API looks good to me - please see comments inline.

File: examples/usage.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    a1 = np.array([
+        (1, 0, 3),
+        (0, 0, 6)
+    ])
+
+    # array conversion
+    a2 = for_backend(a1, SPARSE_GCXS)
+
+    # format determination
+    print("a2 is", get_backend(a2))
```
[the remark this answers is not in the document]
Comment (uellue) [reply examples/usage.py:18]: Good point!

File: src/sparseconverter/__init__.py (hunk 1 above)
Comment (uellue) [reply src/sparseconverter/__init__.py:857]: Good point, thx!

File: src/sparseconverter/__init__.py [hunk 3]
```diff
@@ -0,0 +1,953 @@
...
+        else:
+            return res
+
+    def _get_lazy(self, item):
+        if item == CUPY:
+            import cupy
+            res = cupy.ndarray
+        elif item in (CUPY_SCIPY_COO, CUPY_SCIPY_CSR, CUPY_SCIPY_CSC):
+            import cupyx.scipy  # noqa: F401
+            res = eval(item)
```
[the remark this answers is not in the document]
Comment (uellue) [reply src/sparseconverter/__init__.py:118]: :+1:

**Pull request merged**